source/time_integrator.template.h

```diff
@@ -251,17 +251,17 @@ namespace ryujin
         U, {}, {}, {}, U_[0], precomputed_[0]);
     hyperbolic_module_->apply_boundary_conditions(U_[0], t + tau);
 
-    /* Step 2: U2 = 3/4 U_old + 1/4 (U1 + tau L(U1)) at time t + tau */
+    /* Step 2: U2 = 3/4 U_old + 1/4 (U1 + tau L(U1)) at time t + 0.5 * tau */
```

Comment:
I had to look the scheme up to understand it. The Butcher tableau of the SSP-RK3 scheme is
```
0   | 0
1   | 1   0
1/2 | 1/4 1/4 0
----+------------
    | 1/6 1/6 2/3
```
I have seen the text with the notation involving 3/4, 1/4 etc, but I find that confusing. Anyway, I think the noew code is correct because we have the function evaluation (`appl_boundary_conditions`) at time `t^n+\Delta t` for preparing the evaluation of `k^{(2)}` in the RK notation, then the present line gives needs to be evaluated at `t^n + 1/2 \Delta t`, which then prepares for the evaluation of `k^{(3)}`, and we eventually evaluate at `t^n+\Delta t`, which is the `k^{(1)}` for the next time step. Since we here store the evaluated solution, the order in which we do things is the one you suggest. I am thus approving.

Reply:
@kronbichler Thanks for checking. Incidentally, confusing notation was the main reason why it was wrong in the first place. :sweat_smile: